### PR TITLE
refactor: remove lifecycle dependency from SoundPlayer

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description

Different activities may have different lifecycle behaviors depending on how well they deal with configuration changes or if they have a ViewModel or not

So now the SoundPlayer will get its own scope, which means that callers need to remember to close() it, and it can be enabled/disabled with a property

this also makes removes the Robolectric dependency from SoundPlayerTest, which should make things faster

Additionally, it prepares for handling audios in the new previewer

## Approach
In the commits

## How Has This Been Tested?

Unit tests (should be non functional) + opened some cards with audio in 31 emulator

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Non-blocking, but one should be rebased once the other is in
* https://github.com/ankidroid/Anki-Android/pull/15254

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 1]
```diff
@@ -637,6 +643,7 @@ abstract class AbstractFlashcardViewer :
             cardFrame!!.removeAllViews()
         }
         destroyWebView(webView) // OK to do without a lock
+        soundPlayer.close()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:646]: I believe this also needs `isInitialized`

**Review by david-allison — CHANGES_REQUESTED** [review]
One change

Looks good: only notes are:

* I'd consider moving this `enable/disable/remove` into a `LifecycleObserver` which references the `SoundPlayer`
* `withCol { xxx(this) } ` is subject to change in the next couple of days, low priority to fixup these calls

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
lgtm - will see about a rebase+push to get it in

**Review by mikehardy — CHANGES_REQUESTED** [review]
ok - with apologies, strings are clear but I'm not 100% sure how to do the alteration to take advantage of context receivers to remove the `this` from inside the withCol's, and am prioritizing merging+releasing things that can go quickly with the time I've got

so this will need a closer look

**Review by david-allison — APPROVED** [review]
This is good to go if there's no conflicts, context receivers are optional cleanup

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/SoundPlayer.kt [hunk 2]
```diff
@@ -307,21 +295,13 @@ class SoundPlayer(
     /** Ensures that only one [playSoundsJob] is running at once */
     private suspend fun playSoundsJob(block: suspend CoroutineScope.() -> Unit) {
         val oldJob = playSoundsJob
-        this.playSoundsJob = scope.launch(Dispatchers.IO) {
```
Comment (Mukuljangir372) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/SoundPlayer.kt:310]: We've removed Dispatchers.IO. Is there any specific reason for that?

Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/SoundPlayer.kt:310]: scope = SupervisorJob() + Dispatchers.IO

**Review by mikehardy — APPROVED** [review]
ok - works for me - code looked okay before just halted on conflicts (resolved now)

**Pull request merged**